koku/api/utils.py

```diff
@@ -197,6 +197,22 @@ def month_end(self, in_date):
             return in_date.replace(microsecond=0, second=0, minute=0, hour=0, day=month_end)
         return in_date.replace(day=month_end)
 
+    def latest_date_in_month(self, in_date):
+        """Last day of giving month date or current date (today)."""
+        if isinstance(in_date, str):
+            in_date = parser.parse(in_date)
+        if in_date.year < self.today.year or in_date.month < self.today.month:
+            latest_month_date = in_date.replace(day=self.days_in_month(in_date)).date()
+        else:
+            latest_month_date = self.now.date()
+        return latest_month_date
+
+    def datetime_from_date(self, in_date):
+        """convert date to datetime."""
+        if isinstance(in_date, str):
+            in_date = parser.parse(in_date).date()
+        return datetime.datetime.fromordinal(in_date.toordinal())
```

Comment:
It's cleaner to use `datetime.combine()` and avoid the conversion to an integer.
```suggestion
        return datetime.datetime.combine(in_date, datetime.time())
```